api/models/userModel.js

```diff
@@ -0,0 +1,33 @@
+const User = require('../schemas/userSchema')
+
+module.exports = userModels = {
+    newUser: async (details ,callback) => {
+        const user = new User(details)
+        try {
+            await user.save()
+            callback(undefined, user)
+        } catch (error) {
+            callback(error,undefined)
+        }
+
+    },
+
+    checkExistingUser: async (details, callback) => {
+        try {
+            const uname = await User.find({ username: details.username })
+            const mail = await User.find({ email: details.email })
```

Comment:
@Adhyan12 In model file add two separate function for username check and email check